### PR TITLE
added poolState asserts to testTakeGTAndLTNeutral

### PR DESCRIPTION
**Pull request by ith-harvey**
Simple PR so Akash can perform his excel based testing

### CODE REVIEW

**Review by MikeHathaway — APPROVED** [review]
LGTM

File: src/_test/ERC20Pool/ERC20PoolLiquidationsArbTake.t.sol [hunk 1]
```diff
@@ -378,7 +381,8 @@ contract ERC20PoolLiquidationsArbTakeTest is ERC20HelperContract {
                kickMomp:          9.721295865031779605 * 1e18,
                totalBondEscrowed: 0.195342779771472726 * 1e18,
                auctionPrice:      19.442591730063559200 * 1e18,
-               debtInAuction:     19.778456451861613480 * 1e18
+               debtInAuction:     19.778456451861613480 * 1e18,
+               thresholdPrice:    9.889482233342512889 * 1e18
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsArbTake.t.sol:421]: We should call `assertBorrower` before the `_arbTake` below

File: src/_test/ERC20Pool/ERC20PoolLiquidationsArbTake.t.sol [hunk 2]
```diff
@@ -466,7 +470,8 @@ contract ERC20PoolLiquidationsArbTakeTest is ERC20HelperContract {
...
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsArbTake.t.sol:520]: We should call `assertBorrower` before the `_arbTake` below

File: src/_test/ERC20Pool/ERC20PoolLiquidationsDepositTake.t.sol [hunk 3]
```diff
@@ -378,7 +381,8 @@ contract ERC20PoolLiquidationsDepositTakeTest is ERC20HelperContract {
                kickMomp:          9.721295865031779605 * 1e18,
                totalBondEscrowed: 0.195342779771472726 * 1e18,
                auctionPrice:      19.442591730063559200 * 1e18,
-               debtInAuction:     19.778456451861613480 * 1e18
+               debtInAuction:     19.778456451861613480 * 1e18,
+               thresholdPrice:    9.889482233342512889 * 1e18
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsDepositTake.t.sol:431]: We should call `assertBorrower` before the `_depositTake` below

File: src/_test/ERC20Pool/ERC20PoolLiquidationsDepositTake.t.sol [hunk 4]
```diff
@@ -466,7 +470,8 @@ contract ERC20PoolLiquidationsDepositTakeTest is ERC20HelperContract {
...
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsDepositTake.t.sol:530]: We should call `assertBorrower` before the `_depositTake` below

File: src/_test/ERC20Pool/ERC20PoolLiquidationsDepositTake.t.sol [hunk 5]
```diff
@@ -662,7 +684,8 @@ contract ERC20PoolLiquidationsDepositTakeTest is ERC20HelperContract {
                 kickMomp:          0,
                 totalBondEscrowed: 0,
                 auctionPrice:      0,
-                debtInAuction:     0
+                debtInAuction:     0,
+                thresholdPrice:    0
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsDepositTake.t.sol:764]: We need an `assertAuction` so we can see the `auctionPrice` on LN 705 below in between the `skip(2 hours);` and the `_assertDepositTakeAuctionInsufficientLiquidityRevert` call

File: src/_test/ERC20Pool/ERC20PoolLiquidationsHeal.t.sol [hunk 6]
```diff
@@ -240,6 +242,23 @@ contract ERC20PoolLiquidationsHealTest is ERC20HelperContract {
 
         // skip ahead so take can be called on the loan
         skip(10 hours);
+
+        _assertAuction(
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsHeal.t.sol:273]: We should call `_assertBorrower` before the `_take` below

File: src/_test/ERC20Pool/ERC20PoolLiquidationsHeal.t.sol [hunk 7]
```diff
@@ -603,6 +627,23 @@ contract ERC20PoolLiquidationsHealTest is ERC20HelperContract {
...
+                bondFactor:        0.01 * 1e18,
+                kickTime:          kickTime,
+                kickMomp:          9.721295865031779605 * 1e18,
+                totalBondEscrowed: 98.533942419792216457 * 1e18,
+                auctionPrice:      0.607580991564486240 * 1e18,
+                debtInAuction:     9_976.561670003961916237 * 1e18,
+                thresholdPrice:    9.977074177773911990 * 1e18
+            })
+        );
+
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsHeal.t.sol:737]: We should call `_assertBorrower` before the `_take` below

File: src/_test/ERC20Pool/ERC20PoolLiquidationsKick.t.sol [hunk 8]
```diff
@@ -165,7 +165,8 @@ contract ERC20PoolLiquidationsKickTest is ERC20HelperContract {
                 kickMomp:          0,
                 totalBondEscrowed: 0,
                 auctionPrice:      0,
-                debtInAuction:     0
+                debtInAuction:     0,
+                thresholdPrice:    9.634254807692307697 * 1e18
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsKick.t.sol]: this `_assertAuction` should be moved to after the `skip(100 days)` on LN 174. Additionally an `_assertBorrower` should also be called there before the `kick`

File: src/_test/ERC20Pool/ERC20PoolLiquidationsKick.t.sol [hunk 9]
```diff
@@ -282,7 +284,8 @@ contract ERC20PoolLiquidationsKickTest is ERC20HelperContract {
...
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsKick.t.sol]: this _assertAuction should be moved to after the skip(100 days) on LN 293. Additionally an _assertBorrower should also be called there before the kick

File: src/_test/ERC20Pool/ERC20PoolLiquidationsKick.t.sol [hunk 10]
```diff
@@ -420,7 +426,8 @@ contract ERC20PoolLiquidationsKickTest is ERC20HelperContract {
...
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsKick.t.sol]: this _assertAuction should be moved to after the skip(100 days) on LN 436. Additionally an _assertBorrower should also be called there before the kick

File: src/_test/ERC20Pool/ERC20PoolLiquidationsKick.t.sol [hunk 11]
```diff
@@ -515,7 +524,8 @@ contract ERC20PoolLiquidationsKickTest is ERC20HelperContract {
...
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsKick.t.sol]: This _assertAuction should be moved to after the skip(100 days) on LN 534. Additionally an _assertBorrower should also be called there before the kick

File: src/_test/ERC20Pool/ERC20PoolLiquidationsTake.t.sol [hunk 12]
```diff
@@ -244,6 +246,22 @@ contract ERC20PoolLiquidationsTakeTest is ERC20HelperContract {
...
+                bondFactor:        0.01 * 1e18,
+                kickTime:          block.timestamp - 2 hours,
+                kickMomp:          9.721295865031779605 * 1e18,
+                totalBondEscrowed: 0.195342779771472726 * 1e18,
+                auctionPrice:      155.540733840508473696 * 1e18,
+                debtInAuction:     19.778456451861613480 * 1e18,
+                thresholdPrice:    9.889329828112590306 * 1e18
+            })
+        );
+
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsTake.t.sol:290]: Additionally an `_assertBorrower` should also be called there before the `_take`

File: src/_test/ERC20Pool/ERC20PoolLiquidationsTake.t.sol [hunk 13]
```diff
@@ -294,6 +331,22 @@ contract ERC20PoolLiquidationsTakeTest is ERC20HelperContract {
...
+                bondFactor:        0.01 * 1e18,
+                kickTime:          block.timestamp - 6 hours,
+                kickMomp:          9.721295865031779605 * 1e18,
+                totalBondEscrowed: 0.195342779771472726 * 1e18,
+                auctionPrice:      9.721295865031779616 * 1e18,
+                debtInAuction:     19.778456451861613480 * 1e18,
+                thresholdPrice:    9.889533035607758374 * 1e18
+            })
+        );
+
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsTake.t.sol:385]: An _assertBorrower should also be called here before the `_take`

File: src/_test/ERC20Pool/ERC20PoolLiquidationsTake.t.sol [hunk 14]
```diff
@@ -420,6 +494,22 @@ contract ERC20PoolLiquidationsTakeTest is ERC20HelperContract {
...
+                bondFactor:        0.01 * 1e18,
+                kickTime:          block.timestamp - 358 minutes,
+                kickMomp:          9.721295865031779605 * 1e18,
+                totalBondEscrowed: 98.533942419792216457 * 1e18,
+                auctionPrice:      9.948520384649726656 * 1e18,
+                debtInAuction:     9_976.561670003961916237 * 1e18,
+                thresholdPrice:    9.976867463138769510 * 1e18
+            })
+        );
+
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsTake.t.sol:558]: An _assertBorrower should also be called here before the _take

File: src/_test/ERC20Pool/ERC20PoolLiquidationsTake.t.sol [hunk 15]
```diff
@@ -470,6 +561,22 @@ contract ERC20PoolLiquidationsTakeTest is ERC20HelperContract {
...
+                bondFactor:        0.01 * 1e18,
+                kickTime:          block.timestamp - 10 hours,
+                kickMomp:          9.721295865031779605 * 1e18,
+                totalBondEscrowed: 98.533942419792216457 * 1e18,
+                auctionPrice:      0.607580991564486240 * 1e18,
+                debtInAuction:     9_976.561670003961916237 * 1e18,
+                thresholdPrice:    9.977074177773911990 * 1e18
+            })
+        );
+
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsTake.t.sol:635]: An _assertBorrower should also be called here before the _take

File: src/_test/ERC20Pool/ERC20PoolLiquidationsTake.t.sol [hunk 16]
```diff
@@ -1191,6 +1322,22 @@ contract ERC20PoolLiquidationsTakeTest is ERC20HelperContract {
...
+                bondFactor:        0.01 * 1e18,
+                kickTime:          block.timestamp - 10 hours,
+                kickMomp:          0,
+                totalBondEscrowed: 81.054302378846351183 * 1e18,
+                auctionPrice:      0,
+                debtInAuction:     8_195.704467159075241912 * 1e18,
+                thresholdPrice:    8.196079597628232153 * 1e18
+            })
+        );
+
```
Comment (ith-harvey) [inline src/_test/ERC20Pool/ERC20PoolLiquidationsTake.t.sol:1468]: An _assertBorrower should also be called here before the _take

File: src/_test/ERC721Pool/ERC721PoolLiquidationsKick.t.sol [hunk 17]
```diff
@@ -173,7 +173,8 @@ contract ERC721PoolLiquidationsKickTest is ERC721HelperContract {
                 kickMomp:          0,
                 totalBondEscrowed: 0,
                 auctionPrice:      0,
-                debtInAuction:     0
+                debtInAuction:     0,
+                thresholdPrice:    9.909519230769230774 * 1e18
```
Comment (ith-harvey) [inline src/_test/ERC721Pool/ERC721PoolLiquidationsKick.t.sol]: This `_assertAuction` should be moved after the `skip` . Assert borrower should also be called

File: src/_test/ERC721Pool/ERC721PoolLiquidationsTake.t.sol [hunk 18]
```diff
@@ -173,7 +173,8 @@ contract ERC721PoolLiquidationsTakeTest is ERC721HelperContract {
                 kickMomp:          0,
                 totalBondEscrowed: 0,
                 auctionPrice:      0,
-                debtInAuction:     0
+                debtInAuction:     0,
+                thresholdPrice:    9.909519230769230774 * 1e18
```
Comment (ith-harvey) [inline src/_test/ERC721Pool/ERC721PoolLiquidationsTake.t.sol]: this assertAuction and An _assertBorrower should also be called between skip and take

File: src/_test/ERC721Pool/ERC721PoolLiquidationsTake.t.sol [hunk 19]
```diff
@@ -262,6 +264,22 @@ contract ERC721PoolLiquidationsTakeTest is ERC721HelperContract {
...
+                bondFactor:        0.01 * 1e18,
+                kickTime:          block.timestamp - 5 hours,
+                kickMomp:          9.917184843435912074 * 1e18,
+                totalBondEscrowed: 0.227287198298417188 * 1e18,
+                auctionPrice:      19.834369686871824160 * 1e18,
+                debtInAuction:     23.012828827714740289 * 1e18,
+                thresholdPrice:    11.506709959118993144 * 1e18
+            })
+        );
+
```
Comment (ith-harvey) [inline src/_test/ERC721Pool/ERC721PoolLiquidationsTake.t.sol:293]: assertBorrower call is needed

File: src/_test/ERC721Pool/ERC721PoolLiquidationsTake.t.sol [hunk 20]
```diff
@@ -427,7 +447,8 @@ contract ERC721PoolLiquidationsTakeTest is ERC721HelperContract {
                 kickMomp:          0,
                 totalBondEscrowed: 0,
                 auctionPrice:      0,
-                debtInAuction:     0
+                debtInAuction:     0,
+                thresholdPrice:    9.909519230769230774 * 1e18
```
Comment (ith-harvey) [inline src/_test/ERC721Pool/ERC721PoolLiquidationsTake.t.sol]: move assertAuction to after skip call, assert borrower should also be there

**Review by ith-harvey — COMMENTED** [review]
Hey Prateek please fix, thank you.

File: src/_test/ERC20Pool/ERC20PoolLiquidationsHeal.t.sol (hunk 6 above)
Comment (prat-gpt) [reply src/_test/ERC20Pool/ERC20PoolLiquidationsHeal.t.sol:273]: added

File: src/_test/ERC20Pool/ERC20PoolLiquidationsHeal.t.sol (hunk 7 above)
Comment (prat-gpt) [reply src/_test/ERC20Pool/ERC20PoolLiquidationsHeal.t.sol:737]: added

**Comment by ith-harvey** [comment]
LGTM, merge it!

**Pull request merged**